runtime/src/secondary_index.rs

```diff
@@ -99,12 +103,125 @@ impl SecondaryIndexEntry for RwLockSecondaryIndexEntry {
+{
+    pub fn new(capacity: usize, excluded_keys: &[L1]) -> Self {
+        Self {
+            capacity,
+            map: Vec::new(),
+            excluded_keys: Vec::from(excluded_keys),
+        }
+    }
+    fn get_map(&self) -> &Vec<(L1, L2)> {
+        &self.map
+    }
+    fn sort_slice_by_value(&mut self, slice_key: &L1) {
+        // Obtain a slice of mutable references to all elements with the same key
+        let mut sub_slice = self
+            .map
+            .split_mut(|(k, _)| *k != *slice_key)
+            .flatten()
+            .collect_vec();
+        // Sort them
+        let mut sorting_buffer = sub_slice.iter().map(|x| *(*x)).collect_vec();
+        sorting_buffer.sort_unstable_by_key(|(_, v)| *v);
+        // Copy back into the list
+        for i in 0..sub_slice.len() {
+            *(sub_slice[i]) = sorting_buffer[i];
+        }
+    }
+    fn update_map(&mut self, key: &L1, value: &L2) {
+        // Check if the value already exists.
+        let existing_value_position = self.map.iter().position(|(_, y)| *y == *value);
+        // Remove it if it does.
+        // Note: Removal maintains sorted order, updating would require a re-sort.
+        // Thus, since we have to search to find the new position anyways,
+        // just throw it away and re-insert as if its a new element.
+        if let Some(position) = existing_value_position {
+            self.map.remove(position);
+        }
+        // If its a new value...
+        else {
+            // Check if the list is full, and if the key is less than the smallest element, if so exit early.
+            if self.map.len() >= self.capacity && self.map[0].0 > *key {
+                return;
+            }
+        };
+        // Is this an excluded key?
+        if self.excluded_keys.contains(key) {
+            return;
+        }
+        // Check if there are more elements in the list with the same key
+        let existing_key_position = self.map.iter().position(|(x, _)| *x == *key);
+        // Find where the new entry goes and insert it.
+        let key_position = match self.map.binary_search_by_key(key, |&(k, _)| k) {
+            Ok(found_position) => found_position,
+            Err(woudbe_position) => woudbe_position,
+        };
+        self.map.insert(key_position, (*key, *value));
+        // If there was indeed more elements with the same key sort them by value
+        if existing_key_position.is_some() {
+            self.sort_slice_by_value(key);
+        }
+        // Prune list of too big
+        while self.map.len() > self.capacity {
+            self.map.remove(0);
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct SecondaryIndexLargestKeys(RwLock<HierarchicalOrderedMap<usize, Pubkey>>);
+impl Default for SecondaryIndexLargestKeys {
+    fn default() -> Self {
+        let container =
+            HierarchicalOrderedMap::<usize, Pubkey>::new(NUM_LARGETS_INDEX_KEYS_CACHED, &[0]);
+        SecondaryIndexLargestKeys(RwLock::from(container))
```

Comment:
```suggestion
        SecondaryIndexLargestKeys(RwLock::new(container))
```
This isn't incorrect, but it did make me stop and wonder "why are we using from here instead of new?"